tests/test_workers/test_tasks/test_utils_cache.py

```diff
@@ -0,0 +1,61 @@
+import time
+from unittest import mock
+
+import pytest
+
+from iib.workers.config import TestingConfig
+from iib.workers.dogpile_cache import skopeo_inspect_should_use_cache
+from iib.workers.tasks import utils
+
+
+@pytest.mark.parametrize(
+    "value, result",
+    [
+        ('docker://with_digest@sha256:93120347593478509347tdsvzkljbn', True),
+        ('docker://without_digest:tag', False),
+    ],
+)
+def test_should_cache(value, result):
+    assert skopeo_inspect_should_use_cache(value) == result
+
+
+@pytest.mark.parametrize(
```

Comment:
The use of parametrize implies that each test case is independent, however, they all rely on each other. So all those tests have to be run serially one after the other.

The other issue is that it is susceptible to race conditions, though highly unlikely as you are assuming the second test is executed before the cache expires.